crates/nu-command/src/commands/range.rs

```diff
@@ -29,37 +26,47 @@ impl WholeStreamCommand for Range {
         "Return only the selected rows."
     }
 
-    fn run_with_actions(&self, args: CommandArgs) -> Result<ActionStream, ShellError> {
+    fn run(&self, args: CommandArgs) -> Result<OutputStream, ShellError> {
         range(args)
     }
 }
 
-fn range(args: CommandArgs) -> Result<ActionStream, ShellError> {
-    let (RangeArgs { area }, input) = args.process()?;
-    let range = area.item;
-    let (from, left_inclusive) = range.from;
-    let (to, right_inclusive) = range.to;
-    let from = from.map(|from| *from as usize).unwrap_or(0).saturating_add(
-        if left_inclusive == RangeInclusion::Inclusive {
+fn range(args: CommandArgs) -> Result<OutputStream, ShellError> {
+    let args = args.evaluate_once()?;
+    let cmd_args = RangeArgs {
+        range: args.req(0)?,
+    };
+
+    let (from, left_inclusive) = cmd_args.range.from;
+    let (to, right_inclusive) = cmd_args.range.to;
+    let from_span = from.span;
+    let to_span = to.span;
+
+    let from = from
+        .map(|from| from.as_usize(from_span))
+        .item
+        .unwrap_or(0)
+        .saturating_add(if left_inclusive == RangeInclusion::Inclusive {
             0
         } else {
             1
-        },
-    );
+        });
+
     let to = to
-        .map(|to| *to as usize)
+        .map(|to| to.as_usize(to_span))
+        .item
         .unwrap_or(usize::MAX)
         .saturating_sub(if right_inclusive == RangeInclusion::Inclusive {
             0
         } else {
             1
         });
 
-    Ok(input
-        .skip(from)
-        .take(to - from + 1)
-        .map(ReturnSuccess::value)
-        .to_action_stream())
+    if from > to {
```

Comment:
if from is greater than to, should we do a reverse range like we do with `echo 10..1`? Maybe something to think about in later PRs.

Reply:
Good idea! I was also thinking about having Python-style negative indices (e.g., -2 would mean second last).

Reply:
Could be maybe generalized to ranger in general, not just the range command.